### PR TITLE
Add MiniMax H3 integration

### PR DESCRIPTION
**Pull request by yamada-sexta**
As mentioned here: https://github.com/NVIDIA/flashdreams/issues/443

This commit adds MiniMax H3 integration to flashdreams.

### CODE REVIEW

**Comment by yamada-sexta** [comment]
Tested with both the  regular [MiniMax H3](https://github.com/MiniMax-AI/MiniMax-H3) and the the [Looping-Sketch LoRA](https://huggingface.co/Inner-Reflections/MiniMax-H3-Looping-Sketch-Anime) on A100.

No audio support yet.

File: integrations/minimax_h3/minimax_h3/model.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            audio_indices=state.audio_indices,
+            text_indices=state.text_indices,
+        )
+        for index, (video_timestep, audio_timestep) in enumerate(
+            zip(video_timesteps, audio_timesteps, strict=True)
+        ):
+            print(
+                f"MiniMax H3 denoise step {index + 1}/{len(video_timesteps)}",
+                flush=True,
+            )
```
Comment (ArielG-NV) [inline integrations/minimax_h3/minimax_h3/model.py]: Prefer to use `loguru` for logging.

File: integrations/minimax_h3/minimax_h3/model.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        return (
+            rows.reshape(
+                -1,
+                channels,
+                state.num_latent_frames,
+                state.latent_height,
+                state.latent_width,
+            )
+            .contiguous()
+            .cpu()
```
Comment (ArielG-NV) [inline integrations/minimax_h3/minimax_h3/model.py]: Why are we copying the data here to the CPU?

Looks like we should keep it on what-ever device currently has the data since tracking the var through the code, it looks like we have a branch in use:
1. save to disk via `_save_latents`, calling `latents.detach().cpu().contiguous()` anyways.
2. ensure data is (and copy if not) on CUDA device via `self._decode_video(cache, latents)`. Code ensuring the latents are on the GPU: `state.set("latents", latents.to("cuda"))`

File: integrations/minimax_h3/minimax_h3/pipeline.py [hunk 3]
```diff
@@ -0,0 +1,778 @@
...
+            cache.resumed_stage = "decode"
+            latents = _load_latents(cache, self.config.model_id)
+        else:
+            denoise_started = time.monotonic()
+            if cache.low_ram:
+                latents = self._generate_low_ram(cache)
+            else:
+                latents = self._generate_standard(cache)
+            cache.denoise_seconds = time.monotonic() - denoise_started
+            _save_latents(cache, self.config.model_id, latents)
```
Comment (ArielG-NV) [inline integrations/minimax_h3/minimax_h3/pipeline.py]: we could make this async since otherwise we are blocking `generate` just to generate a cache file we don't immediately need every time we have a new cache?

File: integrations/minimax_h3/minimax_h3/pipeline.py [hunk 4]
```diff
@@ -0,0 +1,778 @@
...
+    def _apply_lora(self, transformer: Any, cache: MiniMaxH3PipelineCache) -> None:
+        if cache.lora is None:
+            return
+        converted = load_lora(
+            transformer,
+            cache.lora,
+            cache.lora_scale,
+            cache.lora_weight_name,
+        )
+        print(f"Loaded LoRA {converted} at scale {cache.lora_scale:g}", flush=True)
```
Comment (ArielG-NV) [inline integrations/minimax_h3/minimax_h3/pipeline.py]: Prefer to use `loguru` for logging.

**Review by ArielG-NV — COMMENTED** [review]
questions/suggestions

File: integrations/minimax_h3/minimax_h3/model.py (hunk 1 above)
Comment (yamada-sexta) [reply integrations/minimax_h3/minimax_h3/model.py]: I have updated this and the other ones

File: integrations/minimax_h3/minimax_h3/model.py (hunk 2 above)
Comment (yamada-sexta) [reply integrations/minimax_h3/minimax_h3/model.py]: Sorry, that was initially a part of the attempt to reduce RAM usage. But you are right, that was unnecessary. I have corrected it.

**Comment by yamada-sexta** [comment]
I tried to use the built in RoPE, but H3 precomputes cos and sin once, shaped [S, 96], then reuses them across its transformer blocks.

H3 rotates only the first 96 channels of each 128-channel attention head; the final 32 channels pass through unchanged.

Therefore, I cannot use the current RoPE effectively. I wonder if you guys are open to changes in the RoPE kernel?

**Comment by ArielG-NV** [comment]
Making changes like adding to the core of flashdreams so that future development of world models is easier for future devleopers is very much encouraged.

This would include changes like adding functionality (knobs for performance) or making alternative implementations of RoPE.

**Comment by yamada-sexta** [comment]
I tested my integration and it seems that the RoPE changes were  not really needed.
